retis/src/module/ovs/bpf/include/ovs_common.h

```diff
@@ -50,10 +49,10 @@ struct {
 
 /* Context saved between the begining and end of ovs_execute_actions calls. */
 struct execute_actions_ctx {
-	struct sk_buff *skb;
+	void *skb;
```

Comment:
Please add a comment about this.

I'm not sure but doing the above might not help in catching casting issues. Not sure either if that is very likely. If we still want to support it, we could use `ifdef __BINDGEN__` for such cases. (Note that I haven't checked if such a thing is defined by bindgen or if we need to manually do it).

Reply:
added `-D__BINDGEN__` to the invocation and kept `struct sk_buff` for the other case

Reply:
Maybe an even better way would be to hide the `__BINDGEN__` use into a macro. Eg.

```c
BINDGEN_PTR(struct sk_buff *, skb)

// Without __BINDGEN__ defined,
struct sk_buff *skb

// With __BINDGEN__ defined,
void *skb
```

That makes the switch automatic for all pointers and adds a clear API when dealing with pointers in the events.

Reply:
what about a more descriptive name? E.g. `BINDING_SKIP_PTR()` ?

Reply:
Yes, the name can be changed. However I'm not sure about "skip", that's describing the internals of how this trick works, not what is provided (here, the ability to define pointers in the UAPI).

Reply:
`BINDING_PTR()` sounds better